include/libtorrent/add_torrent_params.hpp

```diff
@@ -297,6 +297,10 @@ namespace libtorrent
 		// url seeds to be added to the torrent (`BEP 17`_).
 		std::vector<std::string> url_seeds;
 
+		// peers to add to the torrent, to be tried to be connected to as
+		// bittorrent peers.
+		std::vector<tcp::endpoint> peers;
```

Comment:
the only way I can think of would be to support this only via the "url" parameter in add_torrent_params. In this case, the magnet URI is parsed internally in the library, which can use new data structures. just as long as they're not user facing